prow/cluster_lib.sh

```diff
@@ -20,12 +20,162 @@
+    unset IFS
+    k_contexts=$(kubectl config get-contexts -o name)
+    for context in $k_contexts; do
+        if [[ "$PILOT_CLUSTER" != "$context" ]]; then
+            kubectl config use-context ${context}
+             
+            kubectl create ns ${NAMESPACE}
+            kubectl create sa ${SERVICE_ACCOUNT} -n ${NAMESPACE}
+            kubectl create clusterrolebinding istio-multi --clusterrole=cluster-admin --serviceaccount=${NAMESPACE}:${SERVICE_ACCOUNT}
+            CLUSTER_NAME=$(kubectl config view --minify=true -o "jsonpath={.clusters[].name}")
+            if [[ "$CLUSTER_NAME" =~ .*"_".* ]]; then
+                # if clustername has '_' set value to stuff after the last '_' due to k8s secret data name limitation
+                CLUSTER_NAME=${CLUSTER_NAME##*_}
+            fi
+            KUBECFG_FILE=${CLUSTERREG_DIR}/${CLUSTER_NAME}
+            gen_kubeconf_from_sa $SERVICE_ACCOUNT $KUBECFG_FILE
+        fi
+    done
+    kubectl config use-context $PILOT_CLUSTER
+}
+
+function delete_cluster () {
+  if [ "${CLUSTER_CREATED}" = true ]; then
+    gcloud container clusters delete ${CLUSTER_NAME}\
+      --zone ${ZONE}\
+      --project ${PROJECT_NAME}\
+      --quiet\
+      || echo "Failed to delete cluster ${CLUSTER_NAME}"
+  fi
+}
 
 function setup_cluster() {
-  kubectl create clusterrolebinding prow-cluster-admin-binding\
-    --clusterrole=cluster-admin\
-    --user="${KUBE_USER}"
+  # use current-context if pilot_cluster not set
+  PILOT_CLUSTER=${PILOT_CLUSTER:-$(kubectl config current-context)}
+
+  unset IFS
+  k_contexts=$(kubectl config get-contexts -o name)
+  for context in $k_contexts; do
+     kubectl config use-context ${context}
+
+     kubectl create clusterrolebinding prow-cluster-admin-binding\
+       --clusterrole=cluster-admin\
+       --user="${KUBE_USER}"
+  done
+  if [[ "$SETUP_CLUSTERREG" == "True" ]]; then
+      setup_clusterreg
+  fi
+  kubectl config use-context $PILOT_CLUSTER
+
+  if [[ "$USE_GKE" == "True" && "$SETUP_CLUSTERREG" == "True" ]]; then
+    ALL_CLUSTER_CIDRS=
+    for cidr in $(gcloud container clusters list --format='value(clusterIpv4Cidr)'); do
+      if [[ "$ALL_CLUSTER_CIDRS" != "" ]]; then
+        ALL_CLUSTER_CIDRS+=','
+      fi
+      ALL_CLUSTER_CIDRS+=$cidr
+    done
+    ALL_CLUSTER_NETTAGS=
+    for net_tag in $(gcloud compute instances list --format=json | jq '.[].tags.items[0]' | tr -d '"'); do
+      if [[ "$ALL_CLUSTER_NETTAGS" =~ .*"$net_tag".* ]]; then
+        # tag isn't unique so don't add
+        echo "$net_tag isn't unique"
+      else
+        if [[ "$ALL_CLUSTER_NETTAGS" != "" ]]; then
+          ALL_CLUSTER_NETTAGS+=','
+        fi
+        ALL_CLUSTER_NETTAGS+=$net_tag
+      fi
+    done
+    gcloud compute firewall-rules create istio-multicluster-test-pods --allow=tcp,udp,icmp,esp,ah,sctp --direction=INGRESS --priority=900 --source-ranges="$ALL_CLUSTER_CIDRS" --target-tags=$ALL_CLUSTER_NETTAGS --quiet
```

Comment:
you are actually running gcloud on istio-testing here, not on the project where the cluster are created.

Reply:
good catch!